### PR TITLE
feat(thumbnails): implement generation of thumbnails

### PR DESCRIPTION
**Pull request by open-dynaMIX**
This commit...

 - implements a `/hook` endpoint for minio notification
 - implements generation and management of thumbnails for uploaded files
 - adds `family` to `File` for grouping Files
 - adds an `mc` service to docker-compose.yml
       This service will create the bucket and setup the events

### CODE REVIEW

File: alexandria/core/models.py [hunk 1]
```diff
@@ -119,6 +121,16 @@ class Document(UUIDModel):
...
+    ORIGINAL = "original"
+    THUMBNAIL = "thumbnail"
+
+    TYPE_CHOICES = (
+        ORIGINAL,
+        THUMBNAIL,
+    )
+    TYPE_CHOICES_TUPLE = ((type_choice, type_choice) for type_choice in TYPE_CHOICES)
+    type = models.CharField(choices=TYPE_CHOICES_TUPLE, max_length=23, default=ORIGINAL)
+    family = models.ForeignKey("self", on_delete=models.CASCADE, null=True)
```
Comment (winged) [inline alexandria/core/models.py:133]: Wouldn't it be cleaner to have `family` named `original` and set `related_name="thumbnails"`? Or what is the family going to be used for?

File: alexandria/core/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    HTTP_403_FORBIDDEN,
+)
+
+from alexandria.core.models import File
+
+
+@pytest.fixture
+def remove_temp_files(settings):
+    yield
+    for i in settings.THUMBNAIL_CACHE_DIR.glob("**/*"):  # pragma: no cover
```
Comment (winged) [inline alexandria/core/tests/test_views.py:16]: Wouldn't it be cleaner to have a `thumbnail_cache` fixture that uses `tempdir` and mocks `settings.THUMBNAIL_CACHE_DIR` for each run?

File: alexandria/core/models.py (hunk 1 above)
Comment (czosel) [reply alexandria/core/models.py:133]: I'd agree, I prefer `family` to the case where we're dealing with trees.

File: alexandria/core/tests/test_views.py (hunk 2 above)
Comment (open-dynaMIX) [reply alexandria/core/tests/test_views.py]: Absolutely! Done.

File: alexandria/core/models.py (hunk 1 above)
Comment (open-dynaMIX) [reply alexandria/core/models.py]: Done as discussed.

**Comment by open-dynaMIX** [comment]
@winged I've evaluated the idea of using a different bucket for the renderings. IMO this will make things unnecessarily complicated. As soon as we decide to also create other renderings besides thumbnails (e.g. doc --> pdf), we may want to create thumbnails for them (or other renderings).

So I'd say we just put everything into the same bucket and accept the fact that we have to sort out requests for files we don't need to process.

**Comment by winged** [comment]
> @winged I've evaluated the idea of using a different bucket for the renderings. IMO this will make things unnecessarily complicated. As soon as we decide to also create other renderings besides thumbnails (e.g. doc --> pdf), we may want to create thumbnails for them (or other renderings).
> 
> So I'd say we just put everything into the same bucket and accept the fact that we have to sort out requests for files we don't need to process.

Fair enough! And should be rather easy to migrate if we change our minds in the future.

Rest of the code is reviewed and LGTM!

**Pull request merged**